packages/core-p2p/lib/server/versions/peer/handlers/transactions.js

```diff
@@ -37,44 +37,43 @@ exports.store = {
    * @return {Hapi.Response}
    */
   async handler(request, h) {
-    const { eligible, notEligible } = transactionPool.checkEligibility(
-      request.payload.transactions,
-    )
-
     const guard = new TransactionGuard(transactionPool)
 
-    for (const ne of notEligible) {
-      guard.invalidate(ne.transaction, ne.reason)
+    const result = await guard.validate(request.payload.transactions)
+
+    if (result.invalid.length > 0) {
```

Comment:
So, if even only 1 trx in the payload is `invalid`, you'll reject all of them?

Reply:
The code before this patch:
```
  async handler(request, h) {
    const { eligible, notEligible } = transactionPool.checkEligibility(
      request.payload.transactions,
    )

    const guard = new TransactionGuard(transactionPool)

    for (const ne of notEligible) {
      guard.invalidate(ne.transaction, ne.reason)
    }

    await guard.validate(eligible)

    if (guard.hasAny('invalid')) {
      return Boom.notAcceptable(
        'Transactions list could not be accepted.',
        guard.errors,
      )
    }
```
and after the patch:
```
    const guard = new TransactionGuard(transactionPool)

    const result = await guard.validate(request.payload.transactions)

    if (result.invalid.length > 0) {
      return Boom.notAcceptable('Transactions list could not be accepted.', guard.errors)
    }
```
So, I am mimicking the behavior as of before the patch. With such big patches that change lots of things it is best to avoid behavioral changes.